hedera-mirror-rest/utils.js

```
@@ -290,84 +301,122 @@ const getIntegerParam = (param, limit = undefined) => {
  * @return {Array} [query, params] Constructed SQL query fragment and corresponding values
  */
-const parseParams = (paramValues, processOpAndValue) => {
+const parseParams = (paramValues, processValue, processQuery, allowMultiple) => {
   if (paramValues === undefined) {
     return ['', []];
   }
-  // We either have a single entry of account filter, or an array (multiple entries)
-  // Convert a single entry into an array to keep the processing consistent
+  // Convert paramValues to a set to remove duplicates
   if (!Array.isArray(paramValues)) {
-    paramValues = [paramValues];
+    paramValues = new Set([paramValues]);
+  } else {
+    paramValues = new Set(paramValues);
   }
   const partialQueries = [];
   let values = [];
   // Iterate for each value of param. For a url '..?q=val1&q=val2', paramValues for 'q' are [val1, val2].
+  const equalValues = new Set();
   for (const paramValue of paramValues) {
     const opAndValue = parseOperatorAndValueFromQueryParam(paramValue);
     if (opAndValue === null) {
       continue;
     }
-    const queryAndValues = processOpAndValue(opAndValue.op, opAndValue.value);
-    if (queryAndValues !== null) {
-      partialQueries.push(queryAndValues[0]);
-      values = values.concat(queryAndValues[1]);
+    const processedValue = processValue(opAndValue.value);
+    //Equal ops have to be processed in bulk at the end to format the IN() correctly.
+    if (opAndValue.op === opsMap.eq && allowMultiple) {
+      equalValues.add(processedValue);
+    } else {
+      const queryAndValues = processQuery(opAndValue.op, processedValue);
+      if (queryAndValues !== null) {
+        partialQueries.push(queryAndValues[0]);
+        values = values.concat(queryAndValues[1]);
+      }
     }
   }
+  if (equalValues.size !== 0) {
+    const queryAndValues = processQuery(opsMap.eq, Array.from(equalValues));
+    partialQueries.push(queryAndValues[0]);
+    values = values.concat(queryAndValues[1]);
+  }
+
```

We should throw an error if the query replacement parameters doesn't equal the number of values.

---

Done, had to make changes to the way credit/debit parsing returns as it never returned a ?.  Also added some tests for this.